src/main/java/core/basesyntax/ShopServiceImpl.java

```diff
@@ -0,0 +1,17 @@
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+            handler.apply(transaction);
```

Comment:
There is no visible error handling in the case that `operationStrategy.getHandler(transaction.getOperation())` returns `null`. It's a good practice to check for `null` and throw a meaningful exception if no handler is found for the given operation.